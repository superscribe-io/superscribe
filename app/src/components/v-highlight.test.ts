import { test, expect } from 'vitest';
import { mount } from '@vue/test-utils';

import VHighlight from './v-highlight.vue';

test('Mount component', () => {
	expect(VHighlight).toBeTruthy();

	const wrapper = mount(VHighlight, {
		props: {
			text: 'This is a nice text',
			query: ['is', 'text'],
		},
	});

	expect(wrapper.html()).toMatchSnapshot();
});

test('query prop', async () => {
	const wrapper = mount(VHighlight, {
		props: {
			text: 'This is a nice text',
			query: 'nice',
		},
	});

	expect(wrapper.find('.highlight').exists()).toBeTruthy();
	expect(wrapper.find('.highlight').text()).toBe('nice');
});
