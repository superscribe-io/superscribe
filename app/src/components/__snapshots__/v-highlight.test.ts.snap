// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Mount component 1`] = `
"Th
<mark data-v-0a28d51c=\\"\\" class=\\"highlight\\">is</mark>

<mark data-v-0a28d51c=\\"\\" class=\\"highlight\\">is</mark>
 a nice
<mark data-v-0a28d51c=\\"\\" class=\\"highlight\\">text</mark>"
`;
